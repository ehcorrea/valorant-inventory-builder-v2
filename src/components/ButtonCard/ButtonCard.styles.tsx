import { Image, ImageProps } from 'expo-image';
import { TouchableOpacity, View } from 'react-native';
import styled, { css } from 'styled-components/native';

export const Button = styled(TouchableOpacity)`
  ${({ theme }) => css`
    aspect-ratio: 9 / 10;
    border-radius: ${theme.spacings.rhvalue(6)}px;
    flex: 1;
    margin: 5px;
    max-height: ${theme.spacings.rhvalue(324)}px;
    max-width: ${theme.spacings.rwvalue(224)}px;
    overflow: hidden;
  `}
`;

export const ContainerImage = styled(View)`
  ${({ theme }) => css`
    background-color: rgba(217, 217, 217, 0.14);
    flex: 1;
    max-height: 80%;
    padding: ${theme.spacings.rhvalue(5)}px;
    width: auto;
  `}
`;

export const CardImage = styled(Image).attrs({
  contentFit: 'contain',
})<ImageProps>`
  height: 100%;
  width: 100%;
`;

export const Footer = styled(View)`
  ${({ theme }) => css`
    background-color: #0f1923;
    flex: 1;
    max-height: 30%;
    width: auto;
    padding: ${theme.spacings.rhvalue(10)}px;
  `}
`;
