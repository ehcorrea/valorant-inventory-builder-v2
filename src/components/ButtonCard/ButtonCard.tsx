import * as S from './ButtonCard.styles';

type ButtonCardProps = {
  children: React.ReactNode;
  onPress: () => void;
  image: string;
  accessibilityLabel: string;
};

export function ButtonCard({
  image,
  accessibilityLabel,
  children,
  onPress,
}: ButtonCardProps) {
  return (
    <S.Button onPress={onPress}>
      <S.ContainerImage>
        <S.CardImage source={image} accessibilityLabel={accessibilityLabel} />
      </S.ContainerImage>
      {children}
    </S.Button>
  );
}

type FooterProps = {
  children: React.ReactNode;
};

ButtonCard.Footer = ({ children }: FooterProps) => (
  <S.Footer>{children}</S.Footer>
);
