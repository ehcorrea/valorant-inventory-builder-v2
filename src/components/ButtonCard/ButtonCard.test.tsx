import { fireEvent, render } from '@/test/utils';
import { ButtonCard } from './ButtonCard';
import { View } from 'react-native';

const mockedOnPress = jest.fn();

const setup = () =>
  render(
    <ButtonCard
      image="placeholder-image.png"
      onPress={mockedOnPress}
      accessibilityLabel="placeholder label"
    >
      <ButtonCard.Footer>
        <View testID="footer children" />
      </ButtonCard.Footer>
    </ButtonCard>
  );

describe('<ButtonCard/>', () => {
  test('render a presseble card with image and children', () => {
    const container = setup();
    const image = container.getByLabelText('placeholder label');
    const children = container.getByTestId('footer children');
    expect(image.props.source).toBe('placeholder-image.png');
    fireEvent.press(image);
    fireEvent.press(children);
    expect(mockedOnPress).toHaveBeenCalledTimes(2);
  });
});
