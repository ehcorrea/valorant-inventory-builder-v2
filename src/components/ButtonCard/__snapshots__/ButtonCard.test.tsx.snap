// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonCard/> render a presseble card with image and children 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "aspectRatio": 0.9,
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "marginBottom": 5,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 5,
      "maxHeight": 554,
      "maxWidth": 466,
      "opacity": 1,
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "rgba(217, 217, 217, 0.14)",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "maxHeight": "100%",
        "width": "auto",
      }
    }
  >
    <Image
      accessibilityLabel="placeholder label"
      contentFit="contain"
      source="placeholder-image.png"
      style={
        {
          "height": "90%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      {
        "backgroundColor": "#0f1923",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "maxHeight": "30%",
        "paddingBottom": 17,
        "paddingLeft": 17,
        "paddingRight": 17,
        "paddingTop": 17,
        "width": "auto",
      }
    }
  >
    <View
      testID="footer children"
    />
  </View>
</View>
`;
